test/validation/api/ipsec/ipsec.c

```diff
@@ -533,17 +535,36 @@ static int recv_pkts_inline(const ipsec_test_part *part,
 
 		ev = odp_queue_deq(suite_context.queue);
 		if (ODP_EVENT_INVALID != ev) {
+			odp_packet_t pkt;
+			int num_pkts = 0;
+
 			CU_ASSERT(odp_event_is_valid(ev) == 1);
 			CU_ASSERT_EQUAL(ODP_EVENT_PACKET,
 					odp_event_types(ev, &subtype));
 			CU_ASSERT_EQUAL(ODP_EVENT_PACKET_IPSEC,
 					subtype);
 			CU_ASSERT(!part->out[i].status.error.sa_lookup);
 
-			pkto[i] = odp_ipsec_packet_from_event(ev);
-			CU_ASSERT(odp_packet_subtype(pkto[i]) ==
-				  ODP_EVENT_PACKET_IPSEC);
-			i++;
+			pkt = odp_ipsec_packet_from_event(ev);
```

Comment:
After incomplete reassembly the events are not of ODP_EVENT_PACKET_IPSEC subtype. Therefore odp_ipsec_packet_from_event() may not be called before checking the reassembly status.

This makes me wonder if the API is still not clear enough and/or consistent enough regarding the handling of incomplete reassembly.

Reply:
I missed this packet type check. Line 565 was to replace this. I'll have this addressed.